doc/adrs/041_add_support_for_tools_preconditions.adoc

```diff
@@ -0,0 +1,87 @@
+= ADR-041 - Add support for tools preconditions
+
+== Context
+
+When a user clicks on a Node/Edge/Diagram background, the Contextual Palette is openned, containing the tools available for this element.
+We only want to display the tools with no precondition or with a precondition returning `true`.
+
+== Decision
+
+=== Frontend
+
+Rename `getToolSectionsQuery` in operation.ts to `getDiagramDescriptionQuery`` (along with the renaming of the associated mutation and results in `DiagramWebSocketContainer``)
+
+Tools are part of the DiagramDescription, they will be computed on-the-fly from now on. Remove `DiagramDescription.getToolSections() and `GQLDiagramDescription#toolSections``
+
+Remove `toolSections` retrieving part from `getDiagramDescriptionQuery` and add it on `ContextualPalette`` openning.
```

Comment:
I think that `getDiagramDescriptionQuery` can be removed entirely, we need the `getToolSectionQuery` to retrieve the tools because we need to retrieve them at some point 😄